### PR TITLE
chore: added why use LCEL doc

### PR DESCRIPTION
**Pull request by bracesproul**
Added a why use LCEL doc (ported from py library)
Sections in the py doc I removed:
`first-class async support:`
`input and output schemas`

### CODE REVIEW

File: docs/extras/expression_language/why.mdx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Why use LCEL?
+
+The LangChain Expression Language was designed from day 1 to **support putting prototypes in production, with no code changes**, from the simplest “prompt + LLM” chain to the most complex chains (we’ve seen folks successfully running in production LCEL chains with 100s of steps). To highlight a few of the reasons you might want to use LCEL:
+
+- first-class support for streaming: when you build your chains with LCEL you get the best possible time-to-first-token (time elapsed until the first chunk of output comes out). For some chains this means eg. we stream tokens straight from an LLM to a streaming output parser, and you get back parsed, incremental chunks of output at the same rate as the LLM provider outputs the raw tokens. We’re constantly improving streaming support, recently we added a [streaming JSON parser](https://twitter.com/LangChainAI/status/1709690468030914584), and more is in the works.
```
Comment (jacoblee93) [inline docs/extras/expression_language/why.mdx:5]: Ah we don't have the streaming JSON parser yet 😬

Comment (bracesproul) [reply docs/extras/expression_language/why.mdx:5]: Ah sorry, I saw the tweet referenced `class JsonFunctionsOutputParser` and assumed that's what this was

Comment (bracesproul) [inline docs/extras/expression_language/why.mdx:5]: Drop, not avail in JS yet
```suggestion
```

Comment (bracesproul) [reply docs/extras/expression_language/why.mdx]: Dropped

**Comment by jacoblee93** [comment]
Thanks!

**Pull request merged**